pages/common/tsc.md

```diff
@@ -0,0 +1,28 @@
+# tsc
+
+> TypeScript compiler.
+> More information: <https://www.typescriptlang.org/docs/handbook/compiler-options.html>.
+
+- Compile a TypeScript file `hello.ts` into a JavaScript file `hello.js`:
+
+`tsc {{foo.ts}}`
+
+- Compile a TypeScript file into JavaScript using a specific target syntax (default is `ES3`):
+
+`tsc --target {{ES5|ES2015|ES2016|ES2017|ES2018|ESNEXT}} {{foo.ts}}`
+
+- Compile a TypeScript file into a JavaScript file with a custom name:
+
+`tsc --outFile {{foo.js}} {{bar.ts}}`
+
+- Get command line options and files from a file:
+
+`tsc @{{args.txt}}`
+
+- Compile a TypeScript project if outdated:
```

Comment:
What does "if outdated" mean ?

Reply:
I don't know this command that well, to be honest; this particular example was pretty much copied verbatim from the documentation.

I take it to mean something similar to other compilers, i.e. only source files that have changed since the last compilation (perhaps validated via some hash or a cache, not sure) will be re-compiled.

Reply:
The [documentation](https://www.typescriptlang.org/docs/handbook/project-references.html#build-mode-for-typescript) doesn't seem to help much in clarifying what they mean, so I'll assume it's indeed the common compiler behavior I described above.

Reply:
Ah I see. I was confused because you mentioned "project" as a whole. Whereas this means, compile only the files in the project which have changed. Perhaps clarify that ?

Reply:
I guess it checks the mtime of the files involved, and only recompiles if the source files have an mtime that's after the destination file's mtime?

Reply:
I have no idea how the check is implemented, but yeah, it's per-file. I'll reword to clarify.

Reply:
@agnivade, @sbrl: I went back and forth and ended up rephrasing the example differently. I would say the "if outdated" part is not the main point to understand about the `--build` option, and it was getting the description too long, so I removed that bit. Here's the diff, from  608a8dc:

```diff
-- Compile a TypeScript project if outdated:
+- Compile all `.ts` files of a TypeScript project defined in a `tsconfig.json` file:
 
 `tsc --build {{tsconfig.json}}
```

WDYT?

Reply:
Nice!

Reply:
Agree